sdk/search/Azure.Search.Documents/src/Models/VectorQuery.cs

```diff
@@ -1,29 +1,38 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
 using System.Collections.Generic;
+using System.Linq;
 using Azure.Core;
 
 namespace Azure.Search.Documents.Models
 {
-    /// <summary>
-    /// The query parameters for vector and hybrid search queries.
-    /// Please note <see cref="VectorQuery"/> is the base class. According to the scenario, a derived class of the base class might need to be assigned here, or this property needs to be casted to one of the possible derived classes.
-    /// The available derived classes include <see cref="VectorizableTextQuery"/> and <see cref="RawVectorQuery"/>.
-    /// </summary>
-    public abstract partial class VectorQuery
+    public partial class VectorQuery : VectorizableQuery
     {
-        /// <summary> Vector Fields of type Collection(Edm.Single) to be included in the vector searched. </summary>
-        public IList<string> Fields { get; internal set; } = new List<string>();
+        /// <summary> Initializes a new instance of VectorQuery. </summary>
+        /// <param name="vector"> The vector representation of a search query. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="vector"/> is null. </exception>
+        public VectorQuery(IReadOnlyList<float> vector)
```

Comment:
We are currently in the process of updating the vector property type to `ROM<float`. I am currently awaiting guidance on this matter, and Michael will be able to provide us with more information by tomorrow regarding the possibility of introducing a Swagger extension in time for us to rely on the generator. If we cannot achieve this in time, I will proceed with the customization. In the interim, I have added a custom constructor (a private constructor to prevent the generation of a public constructor with an `IEnumerable<float>` vector type). I intend to remove this private constructor once I receive the necessary guidance.

Reply:
Yes, Shivangi's correct here that we're breaking the rules because vectors are being treated as their own type rather than a sequence of values.  We're going to be representing them as `ROM<float>` and started conversations with the TypeSpec and generator folks to get first class representations of these in our specs (mostly likely something like `model EmbeddingVector<T extends numeric = float32> is Array<T> { };` in TypeSpec azure-core and `"x-ms-embedding-vector": true` in swagger).

Reply:
https://github.com/Azure/autorest.csharp/pull/3875 tracking

Reply:
Alpha version of the generator to test it out will merge once we have E2E test confirmation https://github.com/Azure/azure-sdk-for-net/pull/39577/files#diff-93a28d9569550c68624a8ad2209a6fae1d4f88237e4b5414eed2ecac4ef8c98fR179

Reply:
Removed custom ctors and updated `IReadOnlyList<float>` to `ReadOnlyMemory<float>` type for vector property.